### DISCUSSION TITLE
Option for no default layout

### DISCUSSION THREAD

**Discussion by jeroenvanrensen** [livewire/livewire#3476] [3. Feature ideas]
Currently, if you're using Livewire full page components, you'll have to give it a layout.

```php
// config/livewire.php

/*
|--------------------------------------------------------------------------
| Layout
|--------------------------------------------------------------------------
| The default layout view that will be used when rendering a component via
| Route::get('/some-endpoint', SomeComponent::class);. In this case the
| the view returned by SomeComponent will be wrapped in "layouts.app"
|
*/

'layout' => 'layouts.app'
```

Wouldn't it be nice if you just could leave it empty, and Livewire would just use the view, without the layout?

I would like to hear your opinions.

Thank you! Jeroen

**Comment by Qirel** [comment]
I don't really see the need for this. The idea with layouts is that all pages have the same, well - layout, to make it easier to maintain. 
Livewire will inject its component to the `{{ $slot }}` variable in the specified layout, or `layouts/app.blade.php` if none is specified (or whatever you put in the config). If you want an empty layout, just create an empty layout, like `layouts/empty.blade.php`, and render its view like

```html
<div>
    {{ $slot }}
</div>
```

Then you can in your config set the view to `layouts.empty` or if you only want to change specific components, you can chain on `layout('layouts.empty')` to the `view()` method in your `render()` method.

**Reply by jeroenvanrensen** [reply]
Yes, but I prefer to specify the layout in my view, like this:

```blade
<x-layout title="Page Title">
    <div>Contents...</div>
</x-layout>
```

**Comment by joshhanley** [comment]
@jeroenvanrensen while it would be nice to specify the layout in the view, it doesn't work with Livewire, because Livewire injects itself onto the first DOM element inside your blade view.

If that were the full app layout, it would be injecting onto the HTML element, which would cause all sorts of issues with morphdom if it tries to change any of the other stuff (like elements in the head tag).

So technically speaking it wouldn't be possible at this stage with the way Livewire is currently structured.

But I actually make use of two layout files, one is the main app layout `app.blade.php` which only contains the html/head/body tags but no actual layout information, that Livewire automatically uses.

And then make a `page` layout component, which I wrap my blade view in, so it allows me to have different page layouts depending on what I am building.

Have a look at this article I wrote about it https://joshhanley.com.au/articles/how-to-structure-your-layout-file-for-livewire

Hope this helps!

**Reply by jeroenvanrensen** [reply]
Okay that makes sense! Thanks for your detailed answer!
